meson.build

```diff
@@ -0,0 +1,116 @@
+project('dosbox-staging', 'c', 'cpp',
+        default_options : ['cpp_std=c++14', 'b_ndebug=if-release'],
+        version : '0.77.0')
+
+
+# compiler flags
+#
+cc = meson.get_compiler('c')
+cxx = meson.get_compiler('cpp')
+
+add_global_arguments('-Weffc++', language : 'cpp')
+if cxx.has_argument('-Wextra-semi')
+    add_global_arguments('-Wextra-semi', language : 'cpp')
+endif
```

Comment:
Can we similarly add: `-Wall -Wextra -Wshadow -pedantic -Wcast-align -Wunused` for `language : 'c'`?

Reply:
Meson adds `-Wall` by default to all targets :)

As for others - I'll look what kind of warnings do they generate. I remember investigating `-pedantic` recently and the warnings we had were not improving things IIRC. I'll look into it.

Reply:
Sounds good. Yeah, just an idea/discussion point. I think we'll be able to flag "groups" (perhaps in the options.txt file), similar to our existing `warnmore` when we really want to scrub new code.

Reply:
Hey, meson provides the functionality for us already :D Addressed in 09d8be13c805cc2131ec879430fd51025c11d760